src/opendkim.cc

```diff
@@ -1271,16 +1269,18 @@ NAN_METHOD(OpenDKIM::Diffheaders)
       }
       Nan::Set(diffObject, hd_old_key, hd_old);
       Nan::Set(diffObject, hd_new_key, hd_new);
-      Nan::Set(out_array, i, diffObject);
+      Nan::Set(out_array, i, diffObject);      
     }
   } else {
     Nan::ThrowTypeError("diffheaders(): too many headers to fit");
     goto clean;
   }
   
   info.GetReturnValue().Set( out_array );
-  
+  free(*out);
```

Comment:
I assume you did this because you found a leak.  Right below here we also call `free(out)`.  One of these is most certainly wrong and could cause a runtime error.  The docs say that `out` is `A pointer to an array of struct dkim_hdrdiff objects that will be allocated by this function`, which begs the question, why are we doing any kind of `malloc()`.

I think we simply need `out` to be declared as `struct dkim_hdrdiff *out = NULL;` passed as `&out`, accessed as and freed as `out[i].hd_old` and freed as `free(out)`.